openshift/official_template.yaml

```diff
@@ -0,0 +1,162 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: log-anomaly-detector
+  namespace: aiops-prod-log-analysis
+objects:
+- apiVersion: v1
+  kind: PersistentVolumeClaim
+  metadata:
+    labels:
+      app: log-anomaly-detector-f20
+    name: log-anomaly-detector-f20
+  spec:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 1Gi
+    storageClassName: ${STORAGE_CLASS}
+- apiVersion: image.openshift.io/v1
+  kind: ImageStream
+  metadata:
+    labels:
+      app: log-anomaly-detector
+    name: log-anomaly-detector
+  spec:
+    lookupPolicy:
+      local: true
+- apiVersion: build.openshift.io/v1
+  kind: BuildConfig
+  metadata:
+    labels:
+      app: log-anomaly-detector
+    name: anomaly-detector
+  spec:
+    output:
+      to:
+        kind: ImageStreamTag
+        name: log-anomaly-detector:latest
+    source:
+      git:
+        ref: master
+        uri: https://github.com/AICoE/log-anomaly-detector.git
+      type: Git
+    strategy:
+      sourceStrategy:
+        from:
+          kind: ImageStreamTag
+          name: python:3.6
+          namespace: openshift
+      type: Source
+    triggers:
+    - imageChange: {}
+      type: ImageChange
+    - type: ConfigChange
+- apiVersion: apps.openshift.io/v1
+  kind: DeploymentConfig
+  metadata:
+    labels:
+      app: log-anomaly-detector-f20
+    name: log-anomaly-detector-f20
+  spec:
+    replicas: 1
+    selector:
+      deploymentconfig: log-anomaly-detector-f20
+    template:
+      metadata:
+        labels:
+          app: log-anomaly-detector-f20
+          deploymentconfig: log-anomaly-detector-f20
+      spec:
+        containers:
+        - name: log-anomaly-detector
+          image: log-anomaly-detector:latest
+          command: ["python"]
+          args: ["app.py", "run", "--config-yaml", "validation_data/.env_config.yaml"]
+          imagePullPolicy: Always
+          resources:
+            limits:
+              cpu: 2
+              memory: 2Gi
+          volumeMounts:
+          - mountPath: /opt/app-root/src/models
+            name: model-storage
+          - name: escerts
+            mountPath: "/opt/app-root/src/certs"
+            readOnly: true
+          - mountPath: /opt/app-root/src/validation_data
+            name: volume-ehbcr
+
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+        schedulerName: default-scheduler
+        securityContext: {}
+        terminationGracePeriodSeconds: 30
+        volumes:
+          - name: escerts
+            secret:
+              secretName: "log-anomaly-detector-odcs"
```

Comment:
This seems like a very specific secret name mapped to a concrete (internal) use case/deployment - it should probably be templated or have a more general name

Reply:
Good point. I'll export this out as a PARAM

Reply:
Fixed it. Done.